contrib/babelfishpg_tds/src/backend/tds/tdsutils.c

```diff
@@ -967,22 +992,35 @@ handle_rename(RenameStmt* rename_stmt)
 }
 
 /*
- * check_babelfish_renamerole_restrictions
+ * check_babelfish_alterrole_restictions
  *
- * Implements following one additional limitation to drop role stmt
+ * Implements following one additional limitation to alter role stmt
  * block renaming an active babelfish role/user
  */
 static void
-check_babelfish_renamerole_restrictions(char *role)
+check_babelfish_alterrole_restictions(char *role, bool allow_alter_role_operation)
 {
+	Oid bbf_master_guest_oid;
+	Oid bbf_tempdb_guest_oid;
+	Oid bbf_msdb_guest_oid;
+	Oid objectId;
 	if (sql_dialect == SQL_DIALECT_TSQL)
 		return;
-	if (is_babelfish_role(role))
+	objectId = get_role_oid(role, true);
+	bbf_master_guest_oid = get_role_oid("master_guest", true);
+	bbf_tempdb_guest_oid = get_role_oid("tempdb_guest", true);
+	bbf_msdb_guest_oid = get_role_oid("msdb_guest", true);
+	if ((OidIsValid(bbf_master_guest_oid)
+			&& OidIsValid(bbf_tempdb_guest_oid)
+			&& OidIsValid(bbf_msdb_guest_oid)
+			&& is_member_of_role(objectId, bbf_master_guest_oid)
+			&& is_member_of_role(objectId, bbf_tempdb_guest_oid)
+			&& is_member_of_role(objectId, bbf_msdb_guest_oid) &&!allow_alter_role_operation) || is_babelfish_role(role))
```

Comment:
logically, original intention of is_babelfish_role was to include everything related to bbf. 
So you could move these membership checking into is_babelfish_role